### PR TITLE
Feat/cli/improve cog aliged levels

### PR DESCRIPTION
**Pull request by jacott**
1. fix alignment levels to not assume powers of two
1. set explicit targetRes (`-tr`) for cog creation
1. use `-tap` 
    (target aligned pixels) align the coordinates of the extent of the output file to the values of the -tr, such that the aligned extent includes the minimum extent.

### CODE REVIEW

File: packages/shared/src/tms/projection.tile.matrix.set.ts [hunk 1]
```diff
@@ -77,6 +77,25 @@ export class ProjectionTileMatrixSet {
...
+     */
+    getTiffResZoom(resX: number, blockFactor = 1): number {
+        // Get best image resolution
+        const { tms } = this;
+        let z = 0;
+        for (; z < tms.zooms.length; ++z) {
+            if (tms.pixelScale(z) <= resX * blockFactor) return z;
+        }
+        if (z == tms.zooms.length) return z - 1;
+        return -1;
```
Comment (blacha) [inline packages/shared/src/tms/projection.tile.matrix.set.ts:96]: do we want to throw here? does anything check for `== -1`?

File: packages/__tests__/src/rounding.ts [hunk 2]
```diff
@@ -15,10 +15,15 @@ export function round(thing: any, z = 8): any {
         if (typeof obj === 'number') {
             return r(obj);
         }
+        if (typeof obj === 'string') {
+            const n = +obj;
```
Comment (blacha) [inline packages/__tests__/src/rounding.ts:19]: Try and avoid `+` Could we just `Number(obj)` ?

File: packages/__tests__/src/rounding.ts [hunk 3]
```diff
@@ -15,10 +15,15 @@ export function round(thing: any, z = 8): any {
...
+            if (isNaN(n)) return obj;
+            return r(n).toString();
+        }
         if (Array.isArray(obj)) {
             return obj.map(recurse);
         }
-        if (obj == null || typeof obj !== 'object') return obj;
+        if (obj == null || obj.constructor !== Object) return obj;
```
Comment (blacha) [inline packages/__tests__/src/rounding.ts:26]: What is this protecting against?

Comment (jacott) [reply packages/__tests__/src/rounding.ts:26]: It stops copying things like logger which we then can't deep equal against

**Pull request merged**